app/src/main/java/com/example/recipeapp/models/User.java

```diff
@@ -12,127 +13,153 @@
+        query.include(User.KEY_INGREDIENT_ARRAY);
+        query.include(User.KEY_INGREDIENTS_STRING);
+        query.include(User.KEY_PROFILE_IMAGE);
+        query.getInBackground(currentUser.getObjectId(), new GetCallback<ParseUser>() {
+            @Override
+            public void done(ParseUser object, ParseException e) {
+                if (e != null) {
+                    Log.i(TAG, "Unable to fetch user!", e);
+                }
+                Log.i(TAG, "Successfully fetched user!");
+                user.parseUser = object;
+            }
+        });
     }
-    //Create a constructor that sets all fields
+
+    //TODO: Create a constructor that sets all fields
     public ParseUser getParseUser() {
-        return user;
+        return parseUser;
     }
 
     public String getFirstName() {
-        return user.getString(KEY_FIRSTNAME);
+        return parseUser.getString(KEY_FIRSTNAME);
     }
 
     public void setFirstName(String firstName) {
-        user.put(KEY_FIRSTNAME, firstName);
+        parseUser.put(KEY_FIRSTNAME, firstName);
     }
 
     public String getLastName() {
-        return user.getString(KEY_LASTNAME);
+        return parseUser.getString(KEY_LASTNAME);
     }
 
     public void setLastName(String lastName) {
-        user.put(KEY_LASTNAME, lastName);
+        parseUser.put(KEY_LASTNAME, lastName);
     }
 
     public ParseFile getProfileImage() {
-        return user.getParseFile(KEY_PROFILE_IMAGE);
+        return parseUser.getParseFile(KEY_PROFILE_IMAGE);
     }
 
     public void setProfileImage(ParseFile image) {
-        user.put(KEY_PROFILE_IMAGE, image);
+        parseUser.put(KEY_PROFILE_IMAGE, image);
     }
 
     public List<User> getFollowerList() {
-        List<User> followerList = user.getList(KEY_FOLLOWER_LIST);
+        List<User> followerList = parseUser.getList(KEY_FOLLOWER_LIST);
         if (followerList == null)
             return new ArrayList<>();
         return followerList;
     }
 
     public void setFollowerList(List<User> followerList) {
-        user.put(KEY_FOLLOWER_LIST, followerList);
+        parseUser.put(KEY_FOLLOWER_LIST, followerList);
     }
 
     public List<User> getFollowingList() {
-        List<User> followingList = user.getList(KEY_FOLLOWING_LIST);
+        List<User> followingList = parseUser.getList(KEY_FOLLOWING_LIST);
         if (followingList == null)
             return new ArrayList<>();
         return followingList;
     }
 
     public void setFollowingList(List<User> followingList) {
-        user.put(KEY_FOLLOWING_LIST, followingList);
+        parseUser.put(KEY_FOLLOWING_LIST, followingList);
     }
 
-    public List<Ingredient> getIngredientList() {
-        List<Ingredient> ingredientList = user.getList(KEY_INGREDIENT_LIST);
+    public String getIngredientsString() {
+        String ingredients = "";
+        List<Ingredient> ingredientList = this.getIngredientArray();
+        for (int i = 0; i < ingredientList.size(); i++) {
```

Comment:
You could consider a [for-each loop](https://www.programiz.com/java-programming/enhanced-for-loop) here but then, of course, figuring out when to not add a comma gets tricky. You can make this even easier on yourself using [join](https://mkyong.com/java/java-how-to-join-list-string-with-commas/)

Reply:
Since I am creating a string out of each ingredient object's name, would I still be able to use _join_?